alt-tab-macos/ui/PreferencesPanel.swift

```diff
@@ -1,110 +1,211 @@
 import Cocoa
 
-class PreferencesPanel: NSPanel, NSTextViewDelegate {
-    var maxScreenUsage: NSTextView?
-    var windowPadding: NSTextView?
-    var cellPadding: NSTextView?
-    var cellBorderWidth: NSTextView?
-    var maxThumbnailsPerRow: NSTextView?
-    var iconSize: NSTextView?
-    var fontHeight: NSTextView?
-    var interItemPadding: NSTextView?
-    var tabKeyCode: NSTextView?
-    var windowDisplayDelay: NSTextView?
+class PreferencesPanel: NSPanel {
+    // ui: base layout
+    let panelWidth = CGFloat(400)
+    let panelHeight = CGFloat(400) // gets auto adjusted to content height
+    let panelPadding = CGFloat(40)
+    let panelWidthToLabelRatio = CGFloat(0.5)
+
+    // ui: preferences elements
+    var maxScreenUsage: NSSlider?
+    var maxThumbnailsPerRow: NSTextField?
+    var iconSize: NSSlider?
+    var fontHeight: NSSlider?
+    var tabKeyCode: NSTextField?
+    var windowDisplayDelay: NSTextField?
     var metaKey: NSPopUpButton?
     var theme: NSPopUpButton?
     var showOnScreen: NSPopUpButton?
-    var inputsMap = [NSTextView: String]()
+
+    var invisibleTextField: NSTextField? // default firstResponder and used for triggering of focus loose
 
     override init(contentRect: NSRect, styleMask style: StyleMask, backing backingStoreType: BackingStoreType, defer flag: Bool) {
-        super.init(contentRect: contentRect, styleMask: style, backing: backingStoreType, defer: flag)
+        let initialRect = NSRect(x: 0, y: 0, width: panelWidth, height: panelHeight)
+        super.init(contentRect: initialRect, styleMask: style, backing: backingStoreType, defer: flag)
         title = Application.name + " Preferences"
+        titlebarAppearsTransparent = true
```

Comment:
What's the motivation for that change? I was curious and check Apple first-party apps to see if it's the new trend. I found these preferences panels:

**Notes:**
![image](https://user-images.githubusercontent.com/106195/68086338-389af000-fe8e-11e9-8abf-f4beb63d06c1.png)

**Voice Memos:**
![image](https://user-images.githubusercontent.com/106195/68086356-70099c80-fe8e-11e9-9ca9-7ae8dbc8f259.png)

Calendar:
![image](https://user-images.githubusercontent.com/106195/68086396-e908f400-fe8e-11e9-833c-31d71b7120bb.png)

**Safari:**
![image](https://user-images.githubusercontent.com/106195/68086405-06d65900-fe8f-11e9-97dc-77420b54f912.png)

**Terminal:**
![image](https://user-images.githubusercontent.com/106195/68086421-413ff600-fe8f-11e9-928b-03110ad66e17.png)

So not very conclusive. I would say most of the apps on my system are non-Apple, and Apple themselves seem to care less and less about consistency, so to be honest I don't mind this change or the previous version. I was just curious to hear your rationale for changing it

Reply:
My bad, I didn't check apple apps but was thinking they were using transparent title bars (recalled that one incorrectly). I think it looks better, a more modern and flat look. I don't also mind to change it back so we are in line with apple native look. You decide.

In my opinion the titles which include icons for sub-sections of the Preferences you have shown do look kinda dated. 😮 Which of course is understandable since it are just the Preferences of an application so I get the priorities here.

Reply:
I'm fine with both so i guess keep it ;p

Reply:
Thank you for the choice 👍 
Made it native again cause of the sheetModal (NSAlert) otherwise looking displaced.